### PR TITLE
feat: migrate univ2

### LINKED ISSUE #746: migrate UniswapV2

### PR DESCRIPTION
**Pull request by greged93**
Migrate the UniswapV2 tests to the `end_to_end` folder, hence utilising Katana.

Time spent on this PR: 1 day

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe):

## What is the current behavior?

Resolves #746 

## What is the new behavior?

- UniswapV2 tests are ran using Katana.

### CODE REVIEW

File: tests/end_to_end/UniswapV2/conftest.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    return await deploy_solidity_contract(
+        "UniswapV2",
+        "UniswapV2Factory",
+        owner.address,
+        caller_eoa=owner.starknet_contract,
+    )
+
+
+@pytest_asyncio.fixture(scope="module")
+async def pair(
```
Comment (ClementWalter) [inline tests/end_to_end/UniswapV2/conftest.py:46]: seems to be unused

**Review by ClementWalter — CHANGES_REQUESTED** [review]
looks good, just minor fixes

File: scripts/utils/kakarot.py [hunk 2]
```diff
@@ -108,10 +107,10 @@ async def deploy_bytecode(**kwargs) -> Tuple[int, int]:
         for event in receipt.events
         if event.from_address == get_deployments()["kakarot"]["address"]
     ]
+    if len(deploy_event) == 0:
```
[the remark this answers is not in the document]
Comment (greged93) [reply scripts/utils/kakarot.py:110]: This just means that the deploying failed. Can be removed once we bump starknet_py, we can use the `execution_status` field.

File: tests/end_to_end/UniswapV2/conftest.py (hunk 1 above)
Comment (greged93) [reply tests/end_to_end/UniswapV2/conftest.py:46]: Indeed it seems like it wasn't used, will remove.

File: scripts/utils/kakarot.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ClementWalter) [reply scripts/utils/kakarot.py:110]: but it's redundant with the check below `len != 1`

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**